go/libraries/doltcore/sqle/database.go

```diff
@@ -886,3 +964,38 @@ func RegisterSchemaFragments(ctx *sql.Context, db Database, root *doltdb.RootVal
 
 	return nil
 }
+
+// StoredProcedureDetailsToString returns a string encoding of sql.StoredProcedureDetails, which may be used to store
+// the details in string format.
+func StoredProcedureDetailsToString(spd sql.StoredProcedureDetails) string {
+	return fmt.Sprintf("%d|%d|%s", spd.CreatedAt.UTC().UnixNano(), spd.ModifiedAt.UnixNano(), spd.CreateStatement)
```

Comment:
I worry this is not good enough. It's totally possible somebody puts a | character into the body of their stored procedure. Not least because they are doing their own crappy storage hacks that they want to use a stored procedure to make easier.

If the modified / created distinction is important enough to track, then we need to either
1) Modify dolt_schemas to add these fields, or 
2) Store them somewhere else, like on the root value instead of in a user visible table.

Honestly I'm not sure which is better. I like storing them in a table because it makes diffing trivial, but maybe we have outgrown this abstraction. Let's bring in Aaron and talk after standup.

Reply:
Another option would be to use the timestamps from the commit tree for these values. That changes the semantics a bit for changes in the working set, obviously.

Reply:
I thought about this a little more, and the way you have this implemented doesn't have the issue I brought up ("|" characters in the stored procedure body).

I still don't like it, though.